Sources/GRPCCore/Internal/TaskGroup+CancellableTask.swift

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension TaskGroup {
+  /// Adds a child task to the group which is individually cancellable.
+  ///
+  /// - Parameter operation: The task to add to the group.
+  /// - Returns: A handle which can be used to cancel the task without cancelling the rest of
+  ///     the group.
+  @inlinable
+  mutating func addCancellableTask(
+    _ operation: @Sendable @escaping () async -> ChildTaskResult
+  ) -> CancellableTaskHandle {
+    let signal = AsyncStream.makeStream(of: Void.self)
```

Comment:
Just a note for future us, we probably want to use a bare continuation (probably even unsafe) here in the future to avoid all the allocs that `AsyncStream` does

Reply:
Agreed. I deliberately avoided that for now to avoid having to worry about double resumptions etc. but we should definitely do this in the future, I think there are plenty of allocations we can save...!